### ISSUE TITLE
Strange error with docker container : libEGL warning: DRI3: xcb_connect failed

### ISSUE THREAD

**Issue by christopheblin** [klokantech/tileserver-gl#197]
Here is a simple script to reproduce the problem :

ubuntu@geoserver:~/tileserver$ ls

=> the directory is empty

ubuntu@geoserver:~/tileserver$ docker run --name tileserver -it -v $(pwd):/data -p 8080:80 klokantech/tileserver-gl
Starting tileserver-gl v2.1.0
Downloading sample data (zurich_switzerland.mbtiles) from https://github.com/klokantech/tileserver-gl/releases/download/v1.3.0/zurich_switzerland.mbtiles
Automatically creating config file for zurich_switzerland.mbtiles
Run with --verbose to see the config file here.
Starting server
Listening at http://[::]:80/
Startup complete

=> everything works fine, using Ctrl+C to exit interactive mode

ubuntu@geoserver:~/tileserver$ docker start -a tileserver
Starting tileserver-gl v2.1.0
No MBTiles specified, using zurich_switzerland.mbtiles
Automatically creating config file for zurich_switzerland.mbtiles
Run with --verbose to see the config file here.
libEGL warning: DRI3: xcb_connect failed
libEGL warning: DRI2: xcb_connect failed
libEGL warning: DRI2: xcb_connect failed
terminate called after throwing an instance of 'std::runtime_error'
  what():  eglInitialize() failed.

/usr/src/app/run.sh: line 8:    11 Aborted                 (core dumped) node /usr/src/app/ -p 80 "$@"
ubuntu@geoserver:~/tileserver$

I have no idea what is going here ...
It seems I can start the container once but the following tries will always crash with this error.

Is this something normal ?

**Comment by DanielCeregatti** [comment]
I'm also getting this error, except not with docker. I've installed tileserver-gl on Debian 9 using npm according to these instructions http://tileserver.readthedocs.io/en/latest/installation.html#npm. I had to add many more packages than the ones listed there before the npm install succeeded, namely:

```apt install pkg-config libjpeg62-turbo-dev libgit-dev libgles2-mesa libegl1-mesa```

Here's what happens when I run tileserver-gl with one of my own mbtiles files:

```
$ tileserver-gl blockgroup.mbtiles
Starting tileserver-gl v2.1.0
Automatically creating config file for blockgroup.mbtiles
WARN: MBTiles not in "openmaptiles" format. Serving raw data only...
Run with --verbose to see the config file here.
libEGL warning: DRI3: xcb_connect failed
libEGL warning: DRI2: xcb_connect failed
libEGL warning: DRI2: xcb_connect failed
terminate called after throwing an instance of 'std::runtime_error'
  what():  eglInitialize() failed.

Aborted
```

I've seen errors like this before. I know that it should just be a simple matter of running an X server before running tileserver-gl. I run this in a terminal as root:

```# Xvfb :0```

And while the above is running, I run this in another terminal:

```$ DISPLAY=:0.0 tileserver-gl blockgroup.mbtiles
Starting tileserver-gl v2.1.0
Automatically creating config file for blockgroup.mbtiles
WARN: MBTiles not in "openmaptiles" format. Serving raw data only...
Run with --verbose to see the config file here.
libEGL warning: DRI2: failed to create any config
terminate called after throwing an instance of 'std::runtime_error'
  what():  eglInitialize() failed.

Aborted
```

I thought the whole point of using EGL was to remove the dependency of running an X server, yet the first error is indicative of not being able to connect to one. The fact that the second error is different leads me to believe that having a running X server satisfies some dependency, but not all. I tried running this too:

```# Xvfb +iglx :0```

But it made no difference.

Thoughts?

**Comment by DanielCeregatti** [comment]
I noticed there was a run.sh with a comment about Xvfb in the repository. I used the arguments for Xvfb there, but ran it directly:

```Xvfb :0 -screen 0 1024x768x24 -ac +extension GLX +render -noreset```

And now I'm able to run tileserver-gl directly from the command line.

**Comment by smellman** [comment]
@christopheblin 
https://github.com/klokantech/tileserver-gl/blob/master/run.sh#L3
Can you try to make your docker container to change "sleep 1" to "sleep 2" and test it?
I think it is machine spec issue.
In my case, I run 5 tileserver-gl instances in one machine with docker-compose, when restart instances this problem was happen in a few instances because the machine are overloaded.

**Comment by polar** [comment]
I get this too. Apparently when you run it with -it , it's okay. But running it from a start or a restart, I get this error.
I've had to do something like: 

docker create -p 8086:80 -v ${PWD}:/data:ro --entrypoint /bin/bash klokantech/tileserver-gl -c 'start-stop-daemon --start --pidfile ~/xvfb.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :99 -screen 0 1024x768x24 -ac +extension GLX +render -noreset;sleep 5;export DISPLAY=:99.0;cd /data;node /usr/src/app/ -p 80 /data/config.json'

Ugg.

**Comment by petrsloup** [comment]
Try with the latest docker image (`docker pull klokantech/tileserver-gl:latest`). I increased the time for the xvfb daemon to start.

**Comment by polar** [comment]
Increasing the sleep time is not the proper way to go. The command is dying because the X server is not yet up. Sleeping longer *may* work, but is not so guaranteed. You have to wait for the X server to come up. I'll create a pull-request to wait for the X server to come up.

**Comment by petrsloup** [comment]
This was fixed (hopefully) with #242

**Comment by KleggerKoder** [comment]
I used 
`Xvfb :1 -screen 0 1024x768x24 </dev/null &
export DISPLAY=":1"`
then started the server from the command line and it worked again
